### DISCUSSION TITLE
Upgrading from agent v5 to v7

### DISCUSSION THREAD

**Discussion by atibdialpad** [DataDog/datadog-agent#17953] [General]
Hi Team, had a query while migrating agent config files over from v5 to v7. We used to use this config in v5
https://github.com/DataDog/dd-agent/blob/0fec665036997d163f82511157efca4a163fd330/datadog.conf.example#L108C1-L109C18
`allow_ipv6: no` but I do not see any such config for v7. I understand that agent 7 is in Go so using the Python Tornado http client config is not relevant but in general how do we disable the agent connecting to datadog sever over ipv6 in v6+ ?

**Comment by sgnn7** [comment]
Hi @atibdialpad,
Currently I don't believe that there is a way to set this configuration on the Agent 6/7 itself. The Agent will try to use whatever available routes/protocols are present and exclusion of a newer protocol (IPv6) is generally unlikely to be desirable. My general suggestion here would be that the IPv6 gets disabled on the host/machine itself, IPv6 DNS entries for outbound comms be filtered, and/or [a proxy be placed as a target for outbound communication](https://docs.datadoghq.com/agent/proxy/?tab=agentv6v7#squid) where [this IP version preference could take place](https://suay.site/?p=1198).

I'll close this question as resolved but feel free to reopen/reply if needed.
